src/instamatic/gui/gui.py

```diff
@@ -36,31 +37,23 @@ def __init__(self, ctrl=None, stream=None, beam_ctrl=None, app=None, log=None):
         self.log = log
 
         self.q = queue.LifoQueue(maxsize=1)
-        self.triggerEvent = threading.Event()
 
         self.module_io = self.app.get_module('io')
 
-        for name, module in self.app.modules.items():
-            try:
-                module.set_trigger(trigger=self.triggerEvent, q=self.q)
-            except AttributeError:
-                pass  # module does not need/accept a trigger
+        for module in self.app.modules.values():
+            if 'q' in get_type_hints(module.__class__):
+                module.q = getattr(module, 'q', self.q)
```

Comment:
This feels a bit tricky, I think you can make this more explicit by checking for the base class:

```suggestion
        for module in self.app.modules.values():
            if isinstance(module, HasQMixin):
                module.q = self.q
```

Also, I'm not entirely sure why we need the getattr here?